provider/core/src/export/mod.rs

```diff
@@ -26,21 +27,25 @@ pub trait DataExporter: Sync {
     ) -> Result<(), DataError>;
 
     /// Function called for singleton markers.
+    ///
     /// Takes non-mut self as it can be called concurrently.
     fn flush_singleton(
         &self,
         marker: DataMarkerInfo,
         payload: &DataPayload<ExportMarker>,
     ) -> Result<(), DataError> {
         self.put_payload(marker, Default::default(), payload)?;
-        self.flush(marker)
+        self.flush(marker, DeduplicationStrategy::Maximal)
```

Comment:
Suggestion: we don't really care about `DeduplicationStrategy` here; we just care about whether the data was given to us in such a way that DryDataProvider is well-defined. Maybe introduce a new struct/enum. Strawperson:

```rust
#[non_exhaustive]
pub struct FlushMetadata {
    pub is_suitable_for_dry_load: bool,
}
```

(yes, I don't normally like booleans, but they seem appropriate when they are associated with a well-defined black-and-white state; I think this is one but I could be wrong)

Reply:
I think passing the deduplication strategy is most powerful and future-proof. Providers might want to store it to expose it to the user later, and collapsing `None` and `RetainBaseLanguages` into `true` doesn't gain us much.

Reply:
And the other part about making a non-exhaustive FlushMetadata so we don't need to keep adding arguments?